stm/stm.ml

```diff
@@ -370,6 +321,12 @@ end = struct (* {{{ *)
   open Printf
 
   let print_dag vcs () =
+
+    (* Due to threading, this will be called from the toplevel with
+       we_are_parsing set to true, as indeed, the toplevel is waiting
+       for input . What a race! XD *)
+    Flags.we_are_parsing := false;
```

Comment:
Then you won't be able to print anything. You are right int that the boolean should be restored after printing.

Or even better, the boolean should be totally obliterated, but that belongs to a different discussion.